lib/filewatch/boot_setup.rb

```diff
@@ -0,0 +1,91 @@
+require "rbconfig"
+
+module FileWatch
```

Comment:
I'm not a big fan of grouping code by time of execution; aggregating in this file all setup code, constant definitions, structs etc forces a lot of jumping back and forth between files to read the code, for little to no benefit.

Also most of this stuff should have documentation on what it is/is used for, specially since it's now moved away from the code using it.